src/main/java/seedu/address/model/person/Photo.java

```diff
@@ -0,0 +1,66 @@
+package seedu.address.model.person;
+
+import java.io.File;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+
+/**
+ * Represents a Person's display picture in the address book
+ * Guarantees: immutable; is valid as declared in (@Link #isValidImage(String))
+ */
+public class Photo {
+    public static final String MESSAGE_PHOTO_CONSTRAINTS =
+            "Person's photo should be in jpeg and preferred to be of 340px x 453px dimension";
+    public static final String MESSAGE_PHOTO_NOT_FOUND = "Error! Photo does not exist!";
+    /*
+     * The first character of the address must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String PHOTO_VALIDATION_REGEX = "([^\\s]+[\\s\\w]*(\\.(?i)(jpg|jpeg|))$)";
+
+    public final String value;
+
+    /**
+     * Validates given photo.
+     *
+     * @throws IllegalValueException if given photo string is invalid.
+     */
+    public Photo(String photo) throws IllegalValueException {
+        photo = photo.trim();
+
+        if (!isValidPhoto(photo)) {
+            throw new IllegalValueException(MESSAGE_PHOTO_CONSTRAINTS);
+        } else {
+            if (!new File(photo).exists()) {
```

Comment:
Should probably check if it is a file or directory. If it exist() is true, it might be a directory.

Reply:
Ahh, ok! thanks Ritchie, will modify it before i merge